pkg/telemetry/phonehome/interceptor.go

```diff
@@ -0,0 +1,65 @@
+package phonehome
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	erroxGRPC "github.com/stackrox/rox/pkg/errox/grpc"
+	"github.com/stackrox/rox/pkg/grpc/authn"
+	grpcError "github.com/stackrox/rox/pkg/grpc/errors"
+	"github.com/stackrox/rox/pkg/grpc/requestinfo"
+	"github.com/stackrox/rox/pkg/sync"
+	"google.golang.org/grpc"
+)
+
+var (
+	mux = &sync.Mutex{}
+)
+
+func (cfg *Config) track(rp *RequestParams) {
+	id := cfg.HashUserAuthID(rp.UserID)
+	for event, funcs := range cfg.interceptors {
+		props := map[string]any{}
```

Comment:
Coming back to the `props` discussion: if you only want a single interceptor to do meaningful work, then `props` should be a retvalue. If you envision interceptor to leave a trace in the same global to their scope `props` object _regardless whether they return true or false_, then it's not the interceptor who should decide on how collisions should be handled.

Reply:
Filtering interceptors in the chain will decide if the event will fire or not.